inbo/etn
# Add test download dataset

This PR does the following:

1. Add unit-tests for `donwload_dataset()`
2. Add `readr` to `Imports` in `DESCRIPTION` as the readr's, function `write_csv()` is used in `download_dataset()`
2. Add check of `limit` as logical in `get_detections()`
3. ~Refactoring, updating and in some cases simplify tests of `get_animals()` and `get_detections()` as not all of them passed~
4. Replace `%\VignetteIndexEntry{Vignette Title}` in the vignettes header should be changed to the real title as mentioned in https://bookdown.org/yihui/rmarkdown-cookbook/package-vignette.html This is needed as we have now two vignettes, otherwise error arises
5. Replace `library(tidyverse)` by explicitly naming the needed packages in vignette `etn_fields.Rmd`  and add them in `Suggests` section of file `DESCRIPTION`, if they are not already imported or suggested
6. Solve other minor problems which come out while running devtools::check(), e.g. the typical note for not finding global variables which is partially solved as always by adding `.data$` to column names in dplyr functions
7. Fix bug described in #138 
8. Fix enhancement issues #139 and #141
